internal/engine/compiler/impl_amd64.go

```diff
@@ -269,13 +269,30 @@ func (c *amd64Compiler) label(label wazeroir.Label) *amd64LabelInfo {
 
 // compileBuiltinFunctionCheckExitCode implements compiler.compileBuiltinFunctionCheckExitCode for the amd64 architecture.
 func (c *amd64Compiler) compileBuiltinFunctionCheckExitCode() error {
-	if err := c.compileCallBuiltinFunction(builtinFunctionIndexCheckExitCode); err != nil {
+	dstReg, err := c.allocateRegister(registerTypeGeneralPurpose)
+	if err != nil {
+		panic("BUG: cannot find free register")
+	}
+
+	c.assembler.CompileMemoryToRegister(amd64.MOVQ,
+		amd64ReservedRegisterForCallEngine, callEngineModuleContextModuleInstanceOffset, dstReg)
+	c.assembler.CompileMemoryToRegister(amd64.MOVQ, dstReg, 0, dstReg)
```

Comment:
this zero should be defined as const like `moduleInstanceGlobalsOffset` and must be tested in compiler_test.go like others